vscode/src/chat/chat-view/context.ts

```diff
@@ -159,6 +175,53 @@ async function getEnhancedContextFused({
     })
 }
 
+async function getEnhancedContextFromRanker({
+    editor,
+    text,
+    providers,
+    contextRanking,
+}: GetEnhancedContextOptions): Promise<ContextItem[]> {
+    return wrapInActiveSpan('chat.enhancedContextRanker', async span => {
+        // Get all possible context items to rank
+        let searchContext = getVisibleEditorContext(editor)
+
+        const embeddingsContextItemsPromise = retrieveContextGracefully(
+            searchEmbeddingsLocal(providers.localEmbeddings, text),
+            'local-embeddings'
+        )
+
+        const localSearchContextItemsPromise = providers.symf
+            ? retrieveContextGracefully(searchSymf(providers.symf, editor, text), 'symf')
+            : []
+
+        const remoteSearchContextItemsPromise = providers.remoteSearch
+            ? await retrieveContextGracefully(
+                  searchRemote(providers.remoteSearch, text),
+                  'remote-search'
+              )
+            : []
+
+        const keywordContextItemsPromise = (async () => [
+            ...(await localSearchContextItemsPromise),
+            ...(await remoteSearchContextItemsPromise),
+        ])()
+
+        const [embeddingsContextItems, keywordContextItems] = await Promise.all([
+            embeddingsContextItemsPromise,
+            keywordContextItemsPromise,
+        ])
+
+        searchContext = searchContext.concat(keywordContextItems).concat(embeddingsContextItems)
+        const editorContext = await getPriorityContext(text, editor, searchContext)
+        const allContext = editorContext.concat(searchContext)
+        if (!contextRanking) {
+            return allContext
+        }
+        const rankedContext = contextRanking.rankContextItems(text, allContext)
```

Comment:
Let's create a sub-span for this step using `wrapInActiveSpan` so we can see in our tracing data how long this takes for our users

```suggestion
        const rankedContext = wrapInActiveSpan('chat.enhancedContextRanker.reranking', () => contextRanking.rankContextItems(text, allContext))
```

Reply:
nice, this would be great to have !!
Added